Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs

```diff
@@ -0,0 +1,331 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using CsvHelper;
+using Microsoft.CodeAnalysis.Sarif;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Serialization;
+using Taxonomy.Common;
+
+namespace Taxonomy.Cwe
+{
+    public class CweTaxonomyGenerator : TaxonomyGenerator
+    {
+        public List<CweCsvRecord> ReadFromCsv(string filePath)
+        {
+            using FileStream input = File.Open(filePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+            using var textReader = new StreamReader(input);
+            using var csvReader = new CsvReader(textReader, CultureInfo.InvariantCulture);
+
+            return csvReader.GetRecords<CweCsvRecord>().ToList();
+        }
+
+        public Weakness_Catalog ReadFromXml(string filePath)
+        {
+            XmlSerializer serializer = new XmlSerializer(typeof(Weakness_Catalog));
+            return (Weakness_Catalog)serializer.Deserialize(new XmlTextReader(filePath));
+        }
+
+        public void SaveCsvToSarif(string filePath)
+        {
+            var records = this.ReadFromCsv(filePath);
+
+            var run = this.ConvertToSarif(records);
+
+            SarifLog log = new SarifLog
+            {
+                Runs = new Run[] { run }
+            };
+
+            File.WriteAllText(filePath + ".sarif", JsonConvert.SerializeObject(log, Newtonsoft.Json.Formatting.Indented));
+        }
+
+        public void SaveXmlToSarif(string sourceFilePath, string targetFilePath)
+        {
+            var results = this.ReadFromXml(sourceFilePath);
+
+            var run = this.ConvertToSarif(results);
+
+            SarifLog log = new SarifLog
+            {
+                Runs = new Run[] { run }
+            };
+
+            File.WriteAllText(targetFilePath, JsonConvert.SerializeObject(log, Newtonsoft.Json.Formatting.Indented));
+        }
+
+        public Run ConvertToSarif(List<CweCsvRecord> records)
+        {
+            IList<ToolComponent> taxonomies = new List<ToolComponent>();
+            ToolComponent cweTaxonomy = new ToolComponent
+            {
+                Name = "CWE",
+                Guid = Constants.Guid.Cwe,
+                Version = "4.4",
```

Comment:
>"4.4" [](http://example.com/codeflow?start=26&length=5)

maybe we should think about this hard-coded items.
for example, if someone uses to parse a 4.3 it would type 4.4.

Reply:
correct this will be a issue that if user dont know to change they may have this string wrong.
also apply to the release date?
how do you think we should fix this.
Should I add in readme.md to state that manual verify and update these two values after generation?

Reply:
we will add to parameter, like:
--version 4.3 --release-date 2020-12-10